Result/Result.swift

```diff
@@ -31,6 +31,15 @@ public enum Result<T, Error: ErrorType>: ResultType, CustomStringConvertible, Cu
 		}
 	}
 	
+	/// Constructs a result from a function that uses `throw`, failing with `Error` if throws, non-autoclosure version
+    	public init?(_ f: () throws -> T?) {
+        	do {
+            		guard let result = try f() else { return nil }
```

Comment:
But that's a different constructor then. I thought this was to work around the fact that you can't use the `@autoclosure` method with a closure.